File: scala/com/kotori316/transformer/gui/GuiHandler.scala
package com.kotori316.transformer.gui

import com.kotori316.transformer.block.{TileSource, TileTrans}
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler

object GuiHandler extends IGuiHandler {
    val instance = this
    final val guiID_trans = 0
    final val guiID_source = 1

    override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int) = {
        val pos = new BlockPos(x, y, z)
        ID match {
            case 0 => new GuiTrans(world.getTileEntity(pos).asInstanceOf[TileTrans], player)
            case 1 => new GuiSource(player, world.getTileEntity(pos).asInstanceOf[TileSource])
            case _ => null
        }
    }

    override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int) = {
        val pos = new BlockPos(x, y, z)
        ID match {
            case 0 => new ContainerTrans(world.getTileEntity(pos).asInstanceOf[TileTrans], player)
            case 1 => new ContainerSource(player, world.getTileEntity(pos).asInstanceOf[TileSource])
            case _ => null
        }
    }
}

File: src/main/java/com/kotori316/transformer/gui/GuiHandler.java
package com.kotori316.transformer.gui;

import net.minecraft.client.Minecraft;
import net.minecraft.client.entity.EntityPlayerSP;
import net.minecraft.client.gui.GuiScreen;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.math.BlockPos;
import net.minecraftforge.fml.network.FMLPlayMessages;

import com.kotori316.transformer.TileSource;

import static com.kotori316.transformer.TileSource.GUI_ID;

public class GuiHandler {
    public static GuiScreen get(FMLPlayMessages.OpenContainer message) {
        BlockPos pos = message.getAdditionalData().readBlockPos();
        ResourceLocation id = message.getId();
        EntityPlayerSP player = Minecraft.getInstance().player;
        if (GUI_ID.equals(id.toString())) {
            TileSource tileSource = ((TileSource) Minecraft.getInstance().world.getTileEntity(pos));
            return new GuiSource(player, tileSource);
        }
        return null;
    }
}
